File: app/src/main/java/com/inspiringteam/reactivecompass/ReactiveSensors/SensorEventListenerWrapper.java
package com.inspiringteam.reactivecompass.ReactiveSensors;

import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;

import io.reactivex.FlowableEmitter;

public class SensorEventListenerWrapper {

    private FlowableEmitter<SensorEvent> emitter;

    public FlowableEmitter<SensorEvent> getEmitter() {
        return emitter;
    }

    public void setEmitter(FlowableEmitter<SensorEvent> emitter) {
        this.emitter = emitter;
    }

    public SensorEventListener create() {
        return new SensorEventListener() {
            @Override
            public void onSensorChanged(SensorEvent sensorEvent) {
                if (getEmitter() != null) {
                    getEmitter().onNext(sensorEvent);
                }
            }

            @Override
            public void onAccuracyChanged(Sensor sensor, int accuracy) {
                if (getEmitter() != null) {
                    // TODO - implement if needed
                }
            }
        };
    }
}

File: app/src/main/java/com/inspiringteam/reactivecompass/ReactiveSensors/SensorEventListenerWrapper.kt
package com.inspiringteam.reactivecompass.ReactiveSensors

import android.hardware.Sensor
import android.hardware.SensorEvent
import android.hardware.SensorEventListener

import io.reactivex.FlowableEmitter

class SensorEventListenerWrapper {

    var emitter: FlowableEmitter<SensorEvent>? = null

    fun create(): SensorEventListener {
        return object : SensorEventListener {
            override fun onSensorChanged(sensorEvent: SensorEvent) {
                if (emitter != null) {
                    emitter!!.onNext(sensorEvent)
                }
            }

            override fun onAccuracyChanged(sensor: Sensor, accuracy: Int) {
                if (emitter != null) {
                    // TODO - implement if needed
                }
            }
        }
    }
}
